desc/vmec.py

```diff
@@ -1338,6 +1339,209 @@ def read_vmec_output(cls, fname):
+            pressure = eq.pressure
+        else:
+            # if kinetic profiles, fit pressure to power series
+            grid = LinearGrid(L=eq.L_grid, axis=True)
+            data = eq.compute(["rho", "p"], grid=grid)
+            rho = grid.compress(data["rho"])
+            p = grid.compress(data["p"])
+            pressure = PowerSeriesProfile.from_values(rho, p, order=eq.L, sym=True)
+        if isinstance(pressure, PowerSeriesProfile) and pressure.sym:
+            f.write("  AM =")  # pressure power series coefficients
+            for am in pressure.params:
+                f.write(" {:+14.8E}".format(am))
+            f.write("\n  PMASS_TYPE = 'power_series'\n")
+        elif isinstance(pressure, PowerSeriesProfile) and not pressure.sym:
+            rho = np.linspace(0, 1, pressure.basis.L + 1)
+            pressure = SplineProfile(values=pressure(rho), knots=rho)
+        if isinstance(pressure, SplineProfile):
+            f.write("  AM_AUX_S =")  # spline knot locations
+            for r in pressure.knots:
+                f.write(" {:+14.8E}".format(r**2))  # s = rho^2
+            f.write("\n  AM_AUX_F =")  # pressure cubic spline values
+            for am in pressure.params:
+                f.write(" {:+14.8E}".format(am))
+            f.write("\n  PMASS_TYPE = 'cubic_spline'\n")
+
+        f.write("!---- Current/Iota Parameters ----\n")
+        if eq.current is None:
+            iota = eq.iota
+            f.write("  NCURR = 0\n")  # rotational transform profile specified
+            if isinstance(iota, PowerSeriesProfile) and iota.sym:
+                f.write("  AI =")  # iota power series coefficients
+                for ai in iota.params:
+                    f.write(" {:+14.8E}".format(ai))
+                f.write("\n  PIOTA_TYPE = 'power_series'\n")
+            elif isinstance(iota, PowerSeriesProfile) and not iota.sym:
+                rho = np.linspace(0, 1, iota.basis.L + 1)
+                iota = SplineProfile(values=iota(rho), knots=rho)
+            if isinstance(iota, SplineProfile):
+                f.write("  AI_AUX_S =")  # spline knot locations
+                for r in iota.knots:
+                    f.write(" {:+14.8E}".format(r**2))  # s = rho^2
+                f.write("\n  AI_AUX_F =")  # iota cubic spline values
+                for ai in iota.params:
+                    f.write(" {:+14.8E}".format(ai))
+                f.write("\n  PIOTA_TYPE = 'cubic_spline'\n")
+        else:
+            current = eq.current
+            f.write("  NCURR = 1\n")  # current profile specified
+            f.write("  CURTOR = {:+14.8E}\n".format(float(current(1))))  # AC scale
+            if isinstance(current, PowerSeriesProfile) and current.sym:
+                f.write("  AC =")  # current power series coefficients
+                for ac in current.params:
+                    f.write(" {:+14.8E}".format(ac))
+                f.write("\n  PCURR_TYPE = 'power_series_I'\n")
+            else:
```

Comment:
Probably shouldn't use bare else here to account for possibly different profile types in the future